.github/workflows/build.yml

```diff
@@ -28,11 +29,26 @@ jobs:
           cache-dependency-path: 'package-lock.json'
       - name: Install dependencies
         run: npm ci
-      - name: Lint code
-        run: npm run lint
       - name: Lint commit message
         if: github.ref != 'refs/heads/main' && github.actor != 'dependabot[bot]'
-        run: npx commitlint --from HEAD~${{ github.event.pull_request.commits }} --to HEAD
+        run: |
+          npx commitlint --from HEAD~${{ github.event.pull_request.commits }} --to HEAD --verbose
```

Comment:
![image](https://github.com/swagger-api/apidom/assets/193286/e7527ce7-e98c-4386-bf59-4a35ef6110b9)


```suggestion
          npx commitlint --from ${{ github.event.pull_request.base.sha }} --to ${{ github.event.pull_request.head.sha }} --verbose
```

Source: https://stackoverflow.com/questions/67307001/how-to-run-commitlint-in-github-workflow-on-every-commit-of-a-push

Reply:
@char0n 

`only commits in scope of the PR will be checked`: IMO this is desired IIANMS; e.g. in [mentioned run](https://github.com/swagger-api/apidom/actions/runs/5565788003/jobs/10166426126?pr=2954) the issue is that a [previous commit](https://github.com/swagger-api/apidom/commit/c46ae6a3b7eb511e38f7576bfc0bff1a8bf4e9ae) NOT in PR scope fails the lint; this commit should have failed the lint but didn't for some reason, but further PRs shouldn't fail because of that..

`We need the base-SHA as well as the head-SHA`: this is true but again IIANM in my experiments this didn't work as possibly the `github.event.pull_request.base.sha` doesn't point to the correct commit, at least in some scenarios. See e.g. [this discussion](https://github.com/orgs/community/discussions/39880) and related [ticket](https://github.com/actions/runner/issues/1689)

I can try experimenting a bit more with the above variables, but again possibly these tests won't cover all scenarios (other commits to base branch, etc), while the proposed solution based on `github.event.pull_request.commits` and PR head commit seems to work in all scenarios